ddtrace/contrib/rq/__init__.py

```diff
@@ -134,37 +136,43 @@ def traced_queue_enqueue_job(rq, pin, func, instance, args, kwargs):
     else:
         resource = func_name
 
-    with pin.tracer.trace(
-        schematize_messaging_operation("rq.queue.enqueue_job", provider="rq", direction=SpanDirection.OUTBOUND),
+    with core.context_with_data(
+        "rq.queue.enqueue_job",
+        span_name=schematize_messaging_operation(
+            "rq.queue.enqueue_job", provider="rq", direction=SpanDirection.OUTBOUND
+        ),
+        pin=pin,
         service=trace_utils.int_service(pin, config.rq),
         resource=resource,
         span_type=SpanTypes.WORKER,
-    ) as span:
-        span.set_tag_str(COMPONENT, config.rq.integration_name)
-
-        # set span.kind to the type of request being performed
-        span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
-
-        span.set_tag_str("queue.name", instance.name)
-        span.set_tag_str("job.id", job.get_id())
-        span.set_tag_str("job.func_name", job.func_name)
-
+        call_key="queue.enqueue_job",
+        tags={
+            COMPONENT: config.rq.integration_name,
+            SPAN_KIND: SpanKind.PRODUCER,
+            "queue.name": instance.name,
+            "job.id": job.get_id(),
+            "job.func_name": job.func_name,
+        },
+    ) as ctx, ctx[ctx["call_key"]] as span:
         # If the queue is_async then add distributed tracing headers to the job
         if instance.is_async and config.rq.distributed_tracing_enabled:
-            HTTPPropagator.inject(span.context, job.meta)
+            core.dispatch("rq.queue.enqueue_job", [ctx, job])
```

Comment:
Can we pass config.rq to this handler. We should move the details of checking if distributing tracing is enabled to the core api. Like how it's handled in _start_span